### PR TITLE
Federated sharing

### PR DESCRIPTION
**Pull request by malkomich**
Parsing of the remote users from the server, added new permissions for federate sharing, and removed references to the share type in GetRemoteShareesOperations (already defined in ShareType).
@davivel

### CODE REVIEW

**Comment by davivel** [comment]
Cool :)

We should extend a bit the automatic tests to cover federated shares. At least ```CreateShareTest``` and  ```GetShareesTest```.

**Comment by malkomich** [comment]
New automatic tests implemented

File: test_client/tests/src/com/owncloud/android/lib/test_project/test/CreateShareTest.java [hunk 1]
```diff
@@ -217,6 +220,82 @@ public void testCreatePrivateShareWithGroup() {
...
+				RemoteOperationResult.ResultCode.SHARE_FORBIDDEN, 
+				result.getCode()
+		);
+		assertTrue(		// error message from server as part of the result
+				result.getData().size() == 1 && 
+				result.getData().get(0) instanceof String
+		);
+		
+		// remote server doesn't exist
+				result = mActivity.createShare(
```
Comment (davivel) [inline test_client/tests/src/com/owncloud/android/lib/test_project/test/CreateShareTest.java]: Weird padding here on the left.

File: test_client/tests/src/com/owncloud/android/lib/test_project/test/GetShareesTest.java [hunk 2]
```diff
@@ -190,6 +191,59 @@ public void testGetRemoteShareesOperation() {
...
+		// search for sharees including "@"
+		GetRemoteShareesOperation getShareesOperation = new GetRemoteShareesOperation("@", 1, 50);
+		RemoteOperationResult result = getShareesOperation.execute(mClient);
+		JSONObject resultItem;
+		JSONObject value;
+		int type;
+		int fedCount = 0;
+		assertTrue(result.isSuccess() && result.getData().size() > 0);
+		try {
+			for (int i=0; i<result.getData().size(); i++) {
```
Comment (davivel) [inline test_client/tests/src/com/owncloud/android/lib/test_project/test/GetShareesTest.java]: Weird indentation here again; take a look to the options in your editor, please.

**Comment by davivel** [comment]
Just a couple of issues with indentation.

**Comment by malkomich** [comment]
Fixed :)

**Comment by davivel** [comment]
:+1:, ready to go as son as Travis finishes.

**Pull request merged**